docs/src/modules/spring/pages/event-sourced-entities.adoc

```diff
@@ -0,0 +1,160 @@
+
+[source,java]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCart.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCart.java[tag=domain]
+----
+<1> Our `ShoppingCart` is fairly simple, being composed only by a `cartId` and a list of line items.
+<2> A `LineItem` represents a single product and the quantity we intend to buy.
+
+NOTE: Above we are taking advantage of the Java `record` to reduce the amount of boilerplate code, but you can use regular classes so long as they can be serialized.
+
+Another fundamental aspect of our entity will be its domain events. For now, we will have 3 different events `ItemAdded`, `ItemRemoved` and `CheckedOut`, defined as below:
+
+[source,java]
+.src/main/java/com/example/shoppingcart/domain/ShoppingCartEvent.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/domain/ShoppingCartEvent.java[tag=events]
+----
+<1> Include logical type name as JSON property "type."
+<2> Declare subtypes of the serializable polymorphic type `ShoppingCartEvent` associating logical names with each.
+<3> The 3 types of event all derive from the same type `ShoppingCartEvent`.
+
+== Implementing behavior
+
+// TODO make a note somewhere that the component is automatically discovered
+
+Now that we have our Entity state defined along with its events, the remaining steps can be summarized as follows:
+
+- declare your entity and pick an entity key (it needs to be unique as it will be used for sharding purposes);
+- define an access point (i.e. a route path) to your entity;
+- implement how each command is handled and which event(s) it generates;
+- provide a handler for each event and how it affects the entity's state.
+
+Let's have a look at what our shopping cart entity will look like for the first 2 steps from the above list:
+
+[source,java]
+.src/main/java/com/example/shoppingcart/ShoppingCartService.java
+----
+include::example$spring-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/ShoppingCartService.java[tag=class]
+----
+<1> Create a class that extends `EventSourcedEntity<S>`, where `S` is the state type this entity will store (i.e. `ShoppingCart`).
+<2> Make sure to annotate such class with `@Entity` and pass in an `entityKey` and an `entityType`.
+<3> Make use of Spring's RequestMapping annotation to define the routes to your entity.
+
+
+NOTE: The entity key `cartId` must match a path parameter (i.e. `cartId`) and such value needs to be unique per entity. On the other hand, the entity type `shopping-cart` is common for all instances of this entity but should be unique across the different entity types.
```

Comment:
Maybe good to mention that the entityType must be stable and can't be changed once the application is deployed in production.